errors.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goacme
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	errtBadCSR       = "urn:acme:error:badCSR"
+	errtBadNonce     = "urn:acme:error:badNonce"
+	errtConnection   = "urn:acme:error:connection"
+	errtDNSSec       = "urn:acme:error:dnssec"
+	errtMalformed    = "urn:acme:error:malformed"
+	errtInternal     = "urn:acme:error:serverInternal"
+	errtTLS          = "urn:acme:error:tls"
+	errtUnauthorized = "urn:acme:error:unauthorized"
+	errtUnknownHost  = "urn:acme:error:unknownHost"
+	errtRateLimited  = "urn:acme:error:rateLimited"
+)
+
+// Errors predefined by the ACME spec.
+var (
+	ErrBadCSR       = &Error{400, errtBadCSR, "CSR is unacceptable"}
+	ErrBadNonce     = &Error{400, errtBadNonce, "Unacceptable anti-replay nonce"}
+	ErrConnection   = &Error{500, errtConnection, "Could not connect to the client for DV"}
+	ErrDNSSec       = &Error{500, errtDNSSec, "Could not validate a DNSSEC signed domain"}
+	ErrMalformed    = &Error{400, errtMalformed, "Request message is malformed"}
+	ErrInternal     = &Error{500, errtInternal, "Internal Server Error"}
+	ErrTLS          = &Error{500, errtTLS, "TLS error during DV"}
+	ErrUnauthorized = &Error{401, errtUnauthorized, "Client lacks sufficient authorization"}
+	ErrUnknownHost  = &Error{500, errtUnknownHost, "Could not resolve domain name"}
+	ErrRateLimited  = &Error{429, errtRateLimited, "Request exceeds rate limit"}
```

Comment:
but `err.Detail` and `err.Code` is "constant" for these predefined errors. Any other error not matching `acmeErrors` map, you will get the actual error from CA server.

WDTY?